### PR TITLE
[#1833] Add frontend tests for filter glob

### LINKED ISSUE #1833: Add Test Cases for Filter Glob User Input
**What feature(s) would you like to see in RepoSense?**

#1810 was fixed by #1827. However, no frontend test cases were added for this PR due to the high priority.
Please add Cypress test cases to ensure this feature works as expected and facilitate future testing.

**Is the feature request related to a problem?**

**If possible, describe the solution**

**If applicable, describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Punpun1643**
Fixes #1833 

## Proposed commit message

```
There is currently no testing done for the new front-end change 
in filter glob related functionalities.

This may result in regression bugs in the future.

Let's add tests for these new frontend changes made for the filter glob functionalities
to ensure this feature works as expected and facilitate future testing.
```

### CODE REVIEW

File: frontend/cypress/tests/codeView/codeView_filterGlob.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+    cy.get('.radio-button--search')
+        .should('be.visible')
+        .click();
+
+    cy.get('#search')
+        .should('have.value', '');
+  });
+
+  it('check request to filter nothing by enter should show all file types', () => {
```
Comment (Zhou-Jiahao-1998) [inline frontend/cypress/tests/codeView/codeView_filterGlob.js:64]: I think you can extend this test to change its input from something to nothing. That way we can be sure that the filter empty input will return all files rather than not doing any changes to the filter result.

File: frontend/cypress/tests/codeView/codeView_filterGlob.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+    cy.get('#submit-button')
+        .type('{enter}');
+
+    // no file should be shown
+    cy.get('#tab-authorship .files')
+        .should('not.be.visible');
+  });
+
+  it('check request to filter invalid glob by clicking should not show any files', () => {
```
Comment (Zhou-Jiahao-1998) [inline frontend/cypress/tests/codeView/codeView_filterGlob.js:124]: I feel that one test to check if click or enter works will do. After that, we can just test one of it. But this way is fine for me as well. Just that it felt a little redundant to do so.

**Review by Zhou-Jiahao-1998 — CHANGES_REQUESTED** [review]
Sorry for the late review! Please look at my comments and let me know if you have any comments/clarification. Drop me a dm on slack if I do not reply within 2 days.

File: frontend/cypress/tests/codeView/codeView_filterGlob.js (hunk 1 above)
Comment (Punpun1643) [reply frontend/cypress/tests/codeView/codeView_filterGlob.js:64]: Yup I agree, will work on this!

File: frontend/cypress/tests/codeView/codeView_filterGlob.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        .should('be.visible')
+        .first()
+        .click();
+
+    cy.get('.radio-button--search')
+        .should('be.visible')
+        .click();
+
+    // the contents of all the files should be visible
+    cy.get('#tab-authorship .files', { timeout: 90000 })
```
[the remark this answers is not in the document]
Comment (Punpun1643) [reply frontend/cypress/tests/codeView/codeView_filterGlob.js:30]: For this, it is time to wait before the `get` timeout. I think it is ok to remove it as well. But I also thought that we could include the `timeout` option in case the loading of content takes some time to get resolved. `90000` is consistent with other previous tests that try to load files at `#tab-authorship` as well. Feel free to let me know what you think :)

File: frontend/cypress/tests/codeView/codeView_filterGlob.js (hunk 2 above)
Comment (Punpun1643) [reply frontend/cypress/tests/codeView/codeView_filterGlob.js:124]: Hmm the reason I include test for click and enter both times for empty inputs and invalid inputs was that I think invalid inputs, we could check whether both click and enter work. 

For empty input, these tests were added because I remember the filter glob wasn't even clickable and enterable in the first place and this means that all files were present. Entering and clicking for empty inputs is to check whether the button and input work as expected. 

I lean a little towards including these tests, but feel free to let me know as well :)

**Review by Zhou-Jiahao-1998 — APPROVED** [review]
LGTM!

File: frontend/cypress/tests/codeView/codeView_filterGlob.js [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+    cy.get('.radio-button--search')
+        .should('be.visible')
+        .click();
+
+    cy.get('#search')
+        .should('have.value', '');
+  });
+
+  it('check request to filter something to nothing by enter should show all file types', () => {
```
Comment (dcshzj) [inline frontend/cypress/tests/codeView/codeView_filterGlob.js:64]: Can this test name be reworded? I'm not sure what it does by reading this description.

File: frontend/cypress/tests/codeView/codeView_filterGlob.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+    // enter on empty input
+    cy.get('#search')
+        .type('{enter}');
+
+    cy.get('#tab-authorship .files', { timeout: 90000 })
+        .should('be.visible');
+  });
+
+  it('check request to filter something to nothing by clicking should show all file types', () => {
```
Comment (dcshzj) [inline frontend/cypress/tests/codeView/codeView_filterGlob.js:94]: Same for this as well, I think you can be more explicit about it originally having something, then later resetting to empty value or something like that.

**Review by dcshzj — CHANGES_REQUESTED** [review]

File: frontend/cypress/tests/codeView/codeView_filterGlob.js (hunk 4 above)
Comment (Punpun1643) [reply frontend/cypress/tests/codeView/codeView_filterGlob.js]: Hi, the descriptions are updated in the new commit

**Comment by Punpun1643** [comment]
@dcshzj thank you for the review! I have made the changes to the test case descriptions.
